exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/list/ListProofHashCodeCalculatorTest.java

```diff
@@ -137,23 +143,43 @@ void visit_ProofRightValue() {
     calculator = createListProofCalculator();
     calculator.visit(root);
 
-    HashCode expectedRootHash = getBranchHashCode(H1, getNodeHashCode(V2));
+    HashCode expectedRootHash = getBranchHashCode(H1, Optional.of(getNodeHashCode(V2)));
 
     assertThat(calculator.getElements(), equalTo(of(1L, V2)));
     assertEquals(expectedRootHash, calculator.getCalculatedRootHash());
   }
 
+  @Test
+  void visit_FullProof3elements() {
+    ListProofBranch root = new ListProofBranch(
+        new ListProofBranch(
+            leafOf(V1),
+            null
```

Comment:
This structure is not correct — there is only a single element, not three. It will normally be rejected by the structure validator. Here is the correct one:
```
     o
   /  \
  o    o
 / \  / \
e   e e  x   
```

Reply:
Will update.